File: CS/Form1.cs
using System.Windows.Forms;
using DevExpress.XtraCharts;

namespace ChartInteractivePoints {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }

        static int lastY = -1;
        static bool isPressed = false;
        static SeriesPoint selectedPoint = null;

        private void chartControl1_ObjectHotTracked(object sender, HotTrackEventArgs e) {
            if(e.HitInfo.SeriesPoint != null)
                selectedPoint = e.HitInfo.SeriesPoint;

            if(selectedPoint != null && isPressed) {
                DiagramCoordinates point = 
                    ((XYDiagram)(sender as ChartControl).Diagram).PointToDiagram(e.HitInfo.HitPoint);

                if(lastY != -1) {
                    VisualRange range = ((XYDiagram)(sender as ChartControl).Diagram).AxisY.VisualRange;
                    double delta = ((double)range.MaxValue - (double)range.MinValue) / 8;

                    if(selectedPoint.Values[0] >= (double)range.MaxValue - delta)
                        range.MaxValue = selectedPoint.Values[0] + delta;

                    selectedPoint.Values[0] = point.NumericalValue;

                    if(point.QualitativeArgument != "")
                        selectedPoint.Argument = point.QualitativeArgument;
                }

                ((ChartControl)sender).RefreshData();
                lastY = e.HitInfo.HitPoint.Y;
                return;
            }

            lastY = -1;
        }

        private void chartControl1_MouseDown(object sender, MouseEventArgs e) {
            isPressed = true;
        }

        private void chartControl1_MouseUp(object sender, MouseEventArgs e) {
            isPressed = false;
        }
    }
}

File: VB/Form1.vb
Imports System.Windows.Forms
Imports DevExpress.XtraCharts

Namespace ChartInteractivePoints

    Public Partial Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Shared lastY As Integer = -1

        Private Shared isPressed As Boolean = False

        Private Shared selectedPoint As SeriesPoint = Nothing

        Private Sub chartControl1_ObjectHotTracked(ByVal sender As Object, ByVal e As HotTrackEventArgs)
            If e.HitInfo.SeriesPoint IsNot Nothing Then selectedPoint = e.HitInfo.SeriesPoint
            If selectedPoint IsNot Nothing AndAlso isPressed Then
                Dim point As DiagramCoordinates = CType(TryCast(sender, ChartControl).Diagram, XYDiagram).PointToDiagram(e.HitInfo.HitPoint)
                If lastY <> -1 Then
                    Dim range As VisualRange = CType(TryCast(sender, ChartControl).Diagram, XYDiagram).AxisY.VisualRange
                    Dim delta As Double =(CDbl(range.MaxValue) - CDbl(range.MinValue)) / 8
                    If selectedPoint.Values(0) >= CDbl(range.MaxValue) - delta Then range.MaxValue = selectedPoint.Values(0) + delta
                    selectedPoint.Values(0) = point.NumericalValue
                    If Not Equals(point.QualitativeArgument, "") Then selectedPoint.Argument = point.QualitativeArgument
                End If

                CType(sender, ChartControl).RefreshData()
                lastY = e.HitInfo.HitPoint.Y
                Return
            End If

            lastY = -1
        End Sub

        Private Sub chartControl1_MouseDown(ByVal sender As Object, ByVal e As MouseEventArgs)
            isPressed = True
        End Sub

        Private Sub chartControl1_MouseUp(ByVal sender As Object, ByVal e As MouseEventArgs)
            isPressed = False
        End Sub
    End Class
End Namespace
